crates/matrix-sdk-base/src/client.rs

```diff
@@ -1169,7 +1169,12 @@ impl BaseClient {
         // because we want to have the push rules in place before we process
         // rooms and their events, but we want to create the rooms before we
         // process the `m.direct` account data event.
-        self.handle_account_data(&response.account_data.events, &mut changes).await;
+        if let Ok(Some(direct_account_data)) =
```

Comment:
I don't think that this is the correct place to do this. Do note that the currently received account data will only be persisted after the following code snippet further bellow:

https://github.com/matrix-org/matrix-rust-sdk/blob/5827bb7ab33b275835f16eacb4c0fecfa79f4a2b/crates/matrix-sdk-base/src/client.rs#L1187-L1188

If the current sync has some new account data, this `handle_account_data()` call will overwrite it with the old one found in the store.

I think that the most correct way would be to apply the direct targets when we initially create the `Room` object, though that might mean that we load the account data from the store multiple times. I can't come up with a more performant way for this right now.